### PR TITLE
fix: market inplay transition issue

### LINKED ISSUE #81: Orders created in an inplay enabled market after its event start time but before flagged as inplay are not handled well
Orders created in an inplay enabled market after its event start time but before the market has been updated to have its inplay flag set to true are not handled correctly.

Orders placed during this period are treated as pre-event orders and can be matched with other pre-event orders and added to/consume pre-event liquidity. 

Such an order, if not fully matched, might also fall through the gaps of event start triggered order cancellation, i.e., resetting the order book, leaving the order potentially orphaned from the book/liquidity pool, unable to be matched, until it is cancelled at settlement. 

The inplay flag is set by crank asap after reaching a market's event start timestamp. While this will usually be quick, it won't be instantaneous, and there is always the chance cranks cannot perform the update in a timely manner, e.g., RPC provider issues. 

Since the flag is set effectively asynchronously, it really shouldn't be used by the protocol as the exclusive means to determine whether a market is inplay or not. A market's event start timestamp should be used.

### PR DESCRIPTION
**Pull request by eoin-betdex**
Fixes #81 

This addresses the issue raised in #81 ,  updating the protocol to not rely only on a market's `inplay` flag to determine whether the market is in play/in running/live yet, but rather will fall back on checking the market's event start timestamp also. 

```Rust
market.inplay || (market.inplay_enabled && market.event_start_timestamp <= now)
```

This change now means that orders placed on or after a market's event start timestamp will be treated properly as inplay orders - triggering an orderbook reset if the market is configured to do so - and will no longer be considered to be included in or matched against pre-event start liquidity.

### CODE REVIEW

File: tests/end-to-end/inplay_market.ts [hunk 1]
```diff
@@ -45,16 +48,28 @@ describe("End to end test of", () => {
...
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+
+    // Create an inplay order before the market inplay flag has been updated
+    let matchingPool;
+    try {
+      matchingPool = await market.getForMatchingPool(1, 3.0);
+      assert.deepEqual(matchingPool, { len: 1, liquidity: 1, matched: 0 });
+      await market.forOrder(1, 4.2, 3.0, purchaser);
+      matchingPool = await market.getForMatchingPool(1, 3.0);
+      assert.deepEqual(matchingPool, { len: 1, liquidity: 0, matched: 0 });
```
Comment (eoin-betdex) [inline tests/end-to-end/inplay_market.ts:62]: When relying only on `inplay`, this assertion will fail with an actual value of 

```
{ len: 2, liquidity: 5.2, matched: 0 }
```

File: tests/end-to-end/inplay_market.ts [hunk 2]
```diff
@@ -45,16 +48,28 @@ describe("End to end test of", () => {
       assert.equal(e.error.errorCode.code, "MarketEventNotStarted");
     }
 
-    // FIXME could be replaced with a call to market.setEventStartTimestampToNow()
-    while (Math.floor(new Date().getTime() / 1000) <= eventStartTimestamp + 5) {
-      await new Promise((resolve) => setTimeout(resolve, 500));
+    // Move start time up to now
+    await market.updateMarketEventStartTimeToNow();
+    await new Promise((resolve) => setTimeout(resolve, 5000));
```
Comment (pawel-betdex) [inline tests/end-to-end/inplay_market.ts:53]: 5000? painful

**Review by pawel-betdex — APPROVED** [review]

**Review by obi-ewan — DISMISSED** [review]
👍 same comment as Pawel about the 5s delay

File: tests/end-to-end/inplay_market.ts (hunk 2 above)
Comment (eoin-betdex) [reply tests/end-to-end/inplay_market.ts:53]: We can probably reduce this, I just tidied up the `FIXME`. The `+ 5` was there before. 
```
while (Math.floor(new Date().getTime() / 1000) <= eventStartTimestamp + 5) {
      await new Promise((resolve) => setTimeout(resolve, 500));
```

**Review by pawel-betdex — APPROVED** [review]

**Review by obi-ewan — APPROVED** [review]

**Review by obi-ewan — APPROVED** [review]

**Review by stuartbetdex — APPROVED** [review]

**Pull request merged**